modules/education/src/main/java/edu/stanford/spezi/modules/education/videos/component/ExpandableVideoSection.kt

```diff
@@ -0,0 +1,195 @@
+package edu.stanford.spezi.modules.education.videos.component
+
+import androidx.compose.animation.AnimatedVisibility
+import androidx.compose.animation.expandVertically
+import androidx.compose.animation.shrinkVertically
+import androidx.compose.foundation.background
+import androidx.compose.foundation.clickable
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.KeyboardArrowDown
+import androidx.compose.material.icons.filled.KeyboardArrowUp
+import androidx.compose.material3.CardDefaults
+import androidx.compose.material3.ElevatedCard
+import androidx.compose.material3.Icon
+import androidx.compose.material3.MaterialTheme
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.tooling.preview.PreviewParameter
+import androidx.compose.ui.tooling.preview.PreviewParameterProvider
+import edu.stanford.spezi.core.design.theme.Colors
+import edu.stanford.spezi.core.design.theme.Sizes
+import edu.stanford.spezi.core.design.theme.Spacings
+import edu.stanford.spezi.core.design.theme.SpeziTheme
+import edu.stanford.spezi.core.design.theme.TextStyles
+import edu.stanford.spezi.core.design.theme.TextStyles.titleLarge
+import edu.stanford.spezi.modules.education.videos.Video
+import edu.stanford.spezi.modules.education.videos.VideoItem
+
+@Composable
+internal fun SectionHeader(
+    text: String?,
+    isExpanded: Boolean,
+    onHeaderClicked: () -> Unit,
+) {
+    Row(
+        modifier = Modifier
+            .fillMaxWidth()
+            .clickable { onHeaderClicked() }
+            .padding(Spacings.medium),
+        verticalAlignment = Alignment.CenterVertically,
+        horizontalArrangement = Arrangement.SpaceBetween
+    ) {
+        text?.let {
+            Text(
+                text = it,
+                style = titleLarge
+            )
+        }
+        ExpandIcon(isExpanded)
+    }
+}
+
+@Composable
+internal fun ExpandableVideoSection(
+    modifier: Modifier = Modifier,
+    title: String?,
+    description: String?,
+    videos: List<Video> = emptyList(),
+    expandedStartValue: Boolean = false,
+    onExpand: () -> Unit = {},
+    onActionClick: (Video) -> Unit = { _ -> },
+) {
+    var expanded by remember { mutableStateOf(expandedStartValue) }
```

Comment:
Shouldn't be needed, `onExpand` updates the ui state which will cause a new composition. Or you can remove `expandedStartValue`, `onExpand` and `isExpanded` property from video section handle it in compose only with `var expanded by rememberSaveable { mutableStateOf(false) }`